keeper/abci.go

```diff
@@ -249,22 +249,38 @@ func (k *Keeper) PreBlockerHandler(txConfig client.TxConfig) sdk.PreBlocker {
 
 // ----- Utilities -----
 
-func computeVoteExtension(info abci.ExtendedCommitInfo) *VoteExtension {
+func (k *Keeper) computeVoteExtension(ctx context.Context, info abci.ExtendedCommitInfo) *VoteExtension {
+	enrolledValidators, _ := k.GetEnrolledValidators(ctx)
+	totalEnrolled := len(enrolledValidators)
+	var enrolledCount int
+
 	var totalPower int64
 	tallies := make(map[string]int64)
 
 	var winner string
 	var winnerPower int64
 	for _, vote := range info.Votes {
-		totalPower += vote.Validator.Power
-
 		if vote.BlockIdFlag != cmtproto.BlockIDFlagCommit {
 			continue
 		}
+
+		enrolled, _ := k.enrolledValidators.Has(ctx, vote.Validator.Address)
+		if enrolled {
+			enrolledCount++
+		}
+
 		if len(vote.VoteExtension) == 0 {
-			continue
+			// If there are enrolled validators, we check if this vote
+			// extension belongs to an enrolled validator, otherwise we skip
+			// them. If there are no enrolled validators, we default to all
+			// validators being enrolled.
+			if totalEnrolled > 0 && !enrolled {
+				continue
+			}
 		}
 
+		totalPower += vote.Validator.Power
+
 		key := string(vote.VoteExtension)
 		tallies[key] += vote.Validator.Power
```

Comment:
should we maybe use an hex encoded/hash rather than raw bytes casted as a string?